Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift

```diff
@@ -0,0 +1,82 @@
+import SwiftUI
+
+struct SitterProfileView: View {
+    // MARK: - Internal interface
+
+    /// View model responsible to manage data from model layer.
+    @ObservedObject var viewModel = SitterProfileViewModel()
+
+    var body: some View {
+        VStack {
+            if isEditingMode {
+                VStack {
+                    EditSitterInformationView(
+                        name: $viewModel.name,
+                        surname: $viewModel.surname,
+                        phone: $viewModel.phone,
+                        bio: $viewModel.bio,
+                        pricePerHour: $viewModel.pricePerHour
+                    )
+
+                    HStack {
+                        Button("Save") {
+                            viewModel.save()
+                            isEditingMode.toggle()
+                        }
+
+                        Spacer()
+
+                        Button("Cancel") {
+                            isEditingMode.toggle()
+                        }
+                    }
+                }
+                .padding()
+                .buttonStyle(CapsuleWithWhiteText())
+                .background(Color.App.purpleLight)
+                .cornerRadius(AppStyle.UIElementConstant.cornerRadius)
+            } else {
+                VStack {
+                    SitterInfoCardView(
+                        name: viewModel.name,
+                        surname: viewModel.surname,
+                        phone: viewModel.phone,
+                        bio: viewModel.bio,
+                        ratePerHour: viewModel.pricePerHour
+                    )
+
+                    Button("Edit") {
+                        isEditingMode.toggle()
+                    }
+                    .disabled(viewModel.name.isEmpty)
```

Comment:
For now, I can save changes with no name, but then I won't be able to change my profile information at all.

Reply:
It shouldn't work like that, you're right, it my mistake. I fixed it.